lib/exfmt/ast/to_algebra.ex

```diff
@@ -501,7 +529,13 @@ defmodule Exfmt.Ast.ToAlgebra do
 
       # Zero arity call
       %{args: []} ->
-        concat(name, "()")
+        case nested_spec_lhs_call?(ctx) || includes_spec_rhs?(ctx) do
```

Comment:
Do we need to hunt through the entire stack here? Seems you could just check the first few as we only care that it's spec_rhs or spec_lhs in the latest few positions.

We could instead pattern match on the stack in this case expression, which I'd prefer this as then the flow control is uniform in this function.

Might be good to split this out, though for this PR I'd rather keep things uniform.